Client/Coordinators/Router/RouteBuilder.swift

```diff
@@ -135,4 +135,25 @@ struct RouteBuilder {
             TelemetryWrapper.recordEvent(category: .action, method: .open, object: .largeTabsOpenUrl)
         }
     }
+
+    func makeRoute(shortcutItem: UIApplicationShortcutItem) -> Route? {
+        guard let shortcutTypeRaw = shortcutItem.type.components(separatedBy: ".").last,
+              let shortcutType = DeeplinkInput.Shortcut(rawValue: shortcutTypeRaw)
+        else { return nil }
+
+        switch shortcutType {
+        case .newTab:
+            return .search(url: nil, isPrivate: false)
+        case .newPrivateTab:
+            return .search(url: nil, isPrivate: true)
```

Comment:
I think those needs another param about `focusLocationField`, otherwise we won't know in this case it needs to be true (and in other Route cases it will need to be false)